src/LDNS.xs

```diff
@@ -2585,6 +2612,34 @@ rr_nsec3param_salt(obj)
     OUTPUT:
         RETVAL
 
+SV *
+rr_nsec3param_hash_name(obj,name)
+    Zonemaster::LDNS::RR::NSEC3PARAM obj;
+    const char *name;
+    INIT:
+        /* Sanity test on owner name */
+        if (ldns_dname_label_count(ldns_rr_owner(obj)) == 0)
+            XSRETURN_UNDEF;
+    CODE:
+    {
+        ldns_rdf *hashed;
+        ldns_rdf *dname;
+
+        dname = ldns_rdf_new_frm_str(LDNS_RDF_TYPE_DNAME, name);
+        if (!dname)
+            XSRETURN_UNDEF;
+
+        hashed = ldns_nsec3_hash_name_frm_nsec3(obj, dname);
```

Comment:
Same thing here: be sure to free `dname` and the memory allocated by `ldns_rdf2str()`.